build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts

```diff
@@ -70,6 +70,33 @@ sourceSets.all {
     }
 }
 
+val writeGitProperties =
+    tasks.register<WriteProperties>("writeGitProperties") {
+        property("git.build.version", project.version)
+        @Suppress("UnstableApiUsage")
+        property(
+            "git.commit.id",
+            providers
+                .exec { commandLine("git", "rev-parse", "HEAD") }
+                .standardOutput
+                .asText
+                .map { it.trim() }
+        )
+        @Suppress("UnstableApiUsage")
+        property(
+            "git.commit.id.abbrev",
+            providers
+                .exec { commandLine("git", "rev-parse", "--short", "HEAD") }
+                .standardOutput
+                .asText
+                .map { it.trim() }
+        )
+
+        destinationFile.set(layout.buildDirectory.file("generated/git/git.properties"))
+    }
+
+tasks.processResources { from(writeGitProperties) }
+
```

Comment:
Moved this up from `com.hedera.hashgraph.sdk.conventions.gradle.kts` where the `com.gorylenko.gradle-git-properties` plugin was doing this before.
I am not sure if this is still needed/desried at all, but if we have it, I think **all** Modules in the project should have these properties added to the Jar they produce consistently.